runtime/src/bank.rs

```diff
@@ -3788,6 +3727,97 @@ impl Bank {
         }
     }
 
+    fn collect_logs(
+        &self,
+        transactions: &[SanitizedTransaction],
+        execution_results: &[TransactionExecutionResult],
+    ) {
+        let transaction_log_collector_config =
+            self.transaction_log_collector_config.read().unwrap();
+        if transaction_log_collector_config.filter == TransactionLogCollectorFilter::None {
+            return;
+        }
+
+        let collected_logs: Vec<_> = execution_results
+            .iter()
+            .zip(transactions)
```

Comment:
I like the general idea. However, as written, we will now loop over the zipped transactions/execution_results pairs twice; once for the original loop in `load_and_execute_transactions()` and once here.

Perhaps `collect_transaction_logs()` could be plugged into the main loop in `load_and_execute_transactions()` directly, push those `Option<_>`'s into a `Vec` and still only grab the log collector write lock once after the loop ? I do see benefit to minimizing acquisitions of that lock since all of the banking threads could be looking to acquire it

Reply:
You're not wrong but adding an extra loop is going to have a negligible impact on performance. There are many instances of looping over transactions + result vectors in the tx pipeline and timing them. Not to mention this is only going to be looped when logs are enabled (typically only on rpc machines). So I think it's worth pulling this out into a separate loop.

Reply:
> Not to mention this is only going to be looped when logs are enabled (typically only on rpc machines). So I think it's worth pulling this out into a separate loop.

Fair enough